lib/pengine/bundle.c

```diff
@@ -1487,6 +1502,40 @@ bundle_print_xml(pe_resource_t *rsc, const char *pre_text, long options,
     free(child_text);
 }
 
+static void
+bundle_output_xml(pe_resource_t *rsc, long options, pcmk__output_t * out)
+{
+    pe__bundle_variant_data_t *bundle_data = NULL;
+    char *child_text = NULL;
+    CRM_CHECK(rsc != NULL, return);
+
+    get_bundle_variant_data(bundle_data, rsc);
+
+    out->begin_list(out, "bundle", NULL, NULL);
+    out->set_str_prop(out, "id", rsc->id);
+    out->set_str_prop(out, "type", container_agent_str(bundle_data->agent_type));
+    out->set_str_prop(out, "image", bundle_data->image);
+    out->set_bool_prop(out, "unique", is_set(rsc->flags, pe_rsc_unique));
+    out->set_bool_prop(out, "managed", is_set(rsc->flags, pe_rsc_managed));
+    out->set_bool_prop(out, "failed", is_set(rsc->flags, pe_rsc_failed));
```

Comment:
The semantic model, and the fact that we have to accommodate "print-as-you-go" formats like plain text, means we can't have set_*_prop() functions that modify a previous message. Instead, this function should directly do whatever's necessary to output a bundle as XML.

Reply:
I'm sorry, it appeared there mistakenly. Those functions are supposed to me in messages.

Reply:
Please take a look at their usage and implementation in the last amend.